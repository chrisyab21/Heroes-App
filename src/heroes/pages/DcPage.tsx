import { PublisherHeroList } from "../components/PublisherHeroList"




export const DcPage = () => {
  return (

    <div className="p-5">
      <h1>DC Comics</h1>
      <PublisherHeroList publisher="DC"></PublisherHeroList>
    </div>

  )
}
