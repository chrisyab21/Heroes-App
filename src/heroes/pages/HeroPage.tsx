import { Navigate, useNavigate, useParams } from "react-router-dom"
import { getHeroeById } from "../helpers/getHeroeById";

import { FC, useMemo } from 'react';

export const HeroPage: FC = () => {

  const params = useParams();

  const { id } = params;


  const hero = useMemo(() => getHeroeById(id), [id]);

  const navigate = useNavigate();

  const onNavigateback = () => {

    if (hero?.publisher === 'Marvel Comics') {
      //Todo:Arreglar que al dar click vuelva a la pagina correcta
      navigate(`/marvel-Local`, { replace: true });
    }

    if (hero?.publisher === 'DC Comics') {
      //Todo:Arreglar que al dar click vuelva a la pagina correcta
      navigate(`/dc-Local`, { replace: true });
    }

  }


  if (!hero) {
    return <Navigate to={'/marvel-local'} />;
  }


  return (
    <>

      <div className="row mt-5">


        <div className="col-4">

          <img
            src={`/assets/heroes/${id}.jpg`}
            alt={hero.superhero}
            className="img-thumbnail  animate__animated animate__fadeInLeft"
          />

        </div>

        <div className="col-8">
          <h3>{!!hero && hero.superhero}</h3>

          <ul className="list-group list-group-flush">

            <li className="list-group-item">
              <b>Alter Ego: </b>{hero.alter_ego}
            </li>
            <li className="list-group-item">
              <b>Publisher: </b>{hero.publisher}
            </li>
            <li className="list-group-item">
              <b>First Appearance: </b>{hero.first_appearance}
            </li>
          </ul>

          <h5 className="mt-3"></h5>
          <p>{hero.characters}</p>

          <button
            className="btn btn-outline-primary"
            onClick={() => onNavigateback()}
          >
            Regresar


          </button>


        </div>

      </div>

    </>
  )
}
