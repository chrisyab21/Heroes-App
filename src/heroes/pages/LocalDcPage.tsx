import { LocalPublisherHeroList } from "../components/locaPublisherHeroList/LocalPublisherHeroList"




export const LocalDcPage = () => {
  return (

    <div className="p-5">
      <h1>DC Comics</h1>
      <LocalPublisherHeroList publisher="DC Comics"></LocalPublisherHeroList>
    </div>

  )
}
