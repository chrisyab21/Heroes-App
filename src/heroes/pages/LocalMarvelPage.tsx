import { LocalPublisherHeroList } from "../components/locaPublisherHeroList/LocalPublisherHeroList"




export const LocalMarvelPage = () => {
  return (

    <>
      <div className="p-5">
        <h1 style={{ color: "white" }}>Marvel Comics</h1>
        <LocalPublisherHeroList publisher="Marvel Comics"></LocalPublisherHeroList>
      </div>
    </>
  )
}
