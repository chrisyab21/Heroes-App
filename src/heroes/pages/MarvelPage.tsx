import { PublisherHeroList } from "../components/PublisherHeroList"



export const MarvelPage = () => {
  return (

    <>
      <div className="p-5">
        <h1 style={{ color: "white" }}>Marvel Comics</h1>
        <PublisherHeroList publisher="Marvel"></PublisherHeroList>
      </div>
    </>
  )
}
