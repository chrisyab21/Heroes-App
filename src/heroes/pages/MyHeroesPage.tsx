import { useContext, useState } from "react";
import { UserHeroList } from "../components/UserHeroList";
import { HeroeForm } from "../components/heroeForm/HeroeForm";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTrashCan } from "@fortawesome/free-solid-svg-icons";
import { UiContext } from "../../context/contexts/UiContext";
import styles from './myHeroesPage.module.css';

export const MyHeroesPage = () => {

  const { setOpenAddHeroeModal, uiState } = useContext(UiContext);

  console.log("Me renderice: ");


  return (

    <>
      <div className={`p-5 d-flex flex-column align-items-start`}>
        <div className={`${styles.header} align-self-start`}>
          <h1 style={{ color: "white" }}>My Hero List</h1>
          <div className={styles.buttonsContainer}>
            <button type="button" className={`btn btn-primary ${styles.btnResponsive} ${styles.addButton}`} onClick={() => setOpenAddHeroeModal(true)}>
              Agregar heroe
            </button>
            <button type="button" className={`btn btn-danger ${styles.btnResponsive} ${styles.deleteBtn}`} onClick={() => {}}>
              <FontAwesomeIcon icon={faTrashCan} />
            </button>
          </div>
        </div>
        <hr></hr>
        <div className={styles.cardContainer}>
          <UserHeroList />
        </div>
      </div>
      {uiState.openAddHeroeModal && <HeroeForm />}
    </>
  )
}
