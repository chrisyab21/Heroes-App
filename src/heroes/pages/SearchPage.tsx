import { FormEvent, useContext, useState } from "react"
import { useForm } from "../../Hooks/useForm"
import { useLocation, useNavigate } from "react-router-dom"
import queryString from "query-string"
import {searchHeroeByName } from "../../auth/firebase/providers"
import { AuthContext } from "../../context/contexts/AuthContext"
import { HeroCard2 } from "../components/heroeCard/HeroCard2"
import { Heroe } from "../../context/contexts/HeroeContext"

type CustomForm = {

  searchText: string,

}




export const SearchPage = () => {


  const { authState } = useContext(AuthContext);

  const navigate = useNavigate();

  const location = useLocation();

  const query = queryString.parse(location.search);

  const { q = '' } = query;

  const qString = typeof q === "string" ? q : '';

  const searchBool = (qString.length == 0);

  //const heroes = getHeroesByName(qString);


  const formulario: CustomForm = {

    searchText: qString,

  }

  const { formState, onInputChange } = useForm<CustomForm>(formulario);


  const [heroes, setHeroes] = useState<any[]>();



  const onSearchSubmit = async (event: FormEvent) => {

    event.preventDefault();

    const {heroes} = await searchHeroeByName(authState.user?.id!, formState.searchText);

    setHeroes(heroes);


    //if (formState.searchText.trim().length <= 1 ) return;

    navigate(`?q=${formState.searchText}`);

  }


  return (
    <>
      <h1>Search</h1>
      <hr />

      <div className="row">

        <div className="col-5">
          <h4>Buscando</h4>
          <hr />
          <form onSubmit={(event) => onSearchSubmit(event)} >
            <input
              aria-label="input-field"
              type="text"
              placeholder="Buscar Heroe"
              className="form-control"
              name="searchText"
              autoComplete="off"
              value={formState.searchText}
              onChange={(event) => onInputChange(event)}
            />

            <button
              className="btn btn-primary mt-1"

            >
              Buscar

            </button>

          </form>

        </div>

        <div className="col-7">
          <h4>Resultados</h4>
          <hr />

          {
            searchBool ? <div className="alert alert-info"> Buscar un Heroe</div>
              : (heroes?.length === 0) && (<div className="alert alert-danger"> No hay un resultado <b>{q}</b> </div>)
          }

          {
            heroes?.map((hero: Heroe) => (
              <HeroCard2
                key={hero.name}
                heroe={{...hero}}
              ></HeroCard2>

            ))

          }

        </div>

      </div>

    </>
  )
}
