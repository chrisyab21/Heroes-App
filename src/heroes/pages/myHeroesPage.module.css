

.cardContainer {

    width: 100%;
    height: fit-content;
    
}

.header{

    display: flex;
    justify-content: space-between;
    width: 100%;
}

.buttonsContainer{
  display: flex;
  justify-content: space-between;


}

.addButton {


}

.deleteBtn{
  margin-left: 15px;


}




/* Button.module.css */
.btnResponsive {
    /* Ajusta estos valores según tus necesidades */
    width: fit-content;
    height: fit-content;
    font-size: 1rem; /* Tamaño de fuente base */
    font-family: 'Roboto', sans-serif;
   
  }
  
  @media (max-width: 575.98px) {
    .btnResponsive {
      font-size: 0.85rem; /* Más pequeño en dispositivos móviles */
    }
  }
  
  @media (min-width: 576px) and (max-width: 767.98px) {
    .btnResponsive {
      font-size: 0.875rem; /* Intermedio en tablets pequeñas */
    }
  }
  
  @media (min-width: 768px) and (max-width: 991.98px) {
    .btnResponsive {
      font-size: 1rem; /* Default para tablets y laptops pequeñas */
    }
  }
  
  @media (min-width: 992px) {
    .btnResponsive {
      font-size: 1.15rem; /* Más grande en pantallas grandes */
    }
  }
  