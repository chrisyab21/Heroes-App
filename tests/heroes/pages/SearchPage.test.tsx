import { render, screen } from "@testing-library/react";
import {SearchPage} from '../../../src/heroes/pages/SearchPage'
import React from "react";
import { MemoryRouter } from "react-router-dom";
describe('Pruebas en <SearchPage/>', () => {
    
    test('Debe de mostrarse correctamente con valores por defecto', () => { 

       const { container}= render(
            <MemoryRouter>
                <SearchPage></SearchPage>
            </MemoryRouter>
        );

        screen.debug();

        expect(container).toMatchSnapshot();

     });


     test('Debe de mostrar a Batman y el input con el valor del QueryString', () => { 

       render(
            <MemoryRouter initialEntries={['/search?q=Batman']}>
                <SearchPage></SearchPage>
            </MemoryRouter>
        );

        const inputValue = screen.getByLabelText('input-field') as HTMLInputElement;

        expect(inputValue.value).toBe('Batman');


         const img = screen.getByRole('img') as HTMLIFrameElement;

         expect(img.src).toContain("/assets/heroes/dc-batman.jpg");


        screen.debug();

     });

     


 });


 test('Debe de mostrar a un error si no se encuentra el heroe (Batman123)', () => { 

    render(
         <MemoryRouter initialEntries={['/search?q=Batman123']}>
             <SearchPage></SearchPage>
         </MemoryRouter>
     );

     const inputValue = screen.getByLabelText('input-field') as HTMLInputElement;

     expect(inputValue.value).toBe('Batman123');

     expect(screen.getByText(`No hay un resultado`))

     screen.debug();

  });