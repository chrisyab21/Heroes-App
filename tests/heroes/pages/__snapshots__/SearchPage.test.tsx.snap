// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage/> Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Buscando
      </h4>
      <hr />
      <form>
        <input
          aria-label="input-field"
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Buscar Heroe"
          type="text"
          value=""
        />
        <button
          class="btn btn-primary mt-1"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        class="alert alert-info"
      >
         Buscar un Heroe
      </div>
    </div>
  </div>
</div>
`;
